coralproject/talk
# [CORL-2892]: allow additional origins for oembed api calls

## What does this PR do?

These changes add in the ability to add additional tenant-wide domains that can make oEmbed calls for single comment embeds.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

This adds `oEmbedAllowedOrigins` to the `embeddedComments` config so that additional tenant-wide domains can be added that are permitted to call the oEmbed API.

## Does this PR introduce any new environment variables or feature flags?

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

You can test this by going into the admin and going to `Advanced` --> `Embedded comments`. Add/remove domains here and see that all functions/saves as expected. If an invalid URL is entered, you should see an error.

You can test that the domains are working by uncommenting and adding a comment id to the oEmbed example in `commentEmbed.html`. Remove the domain you are calling from from the site config, and see that the oEmbed throws an unauthorized error. Add the domain you are calling from to the new oEmbed api calls configuration and see that you no longer get an unauthorized error and that the domain is now permitted.

## Where any tests migrated to React Testing Library?



## How do we deploy this PR?
